lib/src/replay/screenshot/screenshot_capturer.dart

```diff
@@ -86,79 +86,135 @@ class ScreenshotCapturer {
 
     final viewId = identityHashCode(renderObject);
 
+    final Completer<ImageInfo?> completer = Completer<ImageInfo?>();
+
     try {
       final srcWidth = renderObject.size.width;
       final srcHeight = renderObject.size.height;
       final pixelRatio =
           _getPixelRatio(srcWidth: srcWidth, srcHeight: srcHeight);
 
-      final ui.Image image = await renderObject.toImage(pixelRatio: pixelRatio);
+      final syncImage = renderObject.toImage(pixelRatio: pixelRatio);
```

Comment:
well this is misleading because its not sync, since returns a future and you don't `await`.